### PR TITLE
cli and sdk tweaks

### PR DESCRIPTION
**Pull request by davidiw**
* cli recursive checks were broken
* ts didn't make it easy to filter for tests
* python was missing a critical instruction on how to use poetry

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/3239)

### CODE REVIEW

File: crates/aptos/src/config/mod.rs [hunk 1]
```diff
@@ -236,7 +236,7 @@ fn find_workspace_config(
                 current_path.push(CONFIG_FOLDER);
                 if current_path.is_dir() {
                     break Ok(current_path);
-                } else if !current_path.pop() && !current_path.pop() {
+                } else if !(current_path.pop() && current_path.pop()) {
```
Comment (movekevin) [inline crates/aptos/src/config/mod.rs:239]: This changes the original condition here. Was this the root cause?

**Review by wrwg — APPROVED** [review]

**Review by banool — APPROVED** [review]
Didn't look too hard at what Kevin pointed out but the rest looks good.

File: crates/aptos/src/config/mod.rs (hunk 1 above)
Comment (davidiw) [reply crates/aptos/src/config/mod.rs:239]: if I give you "/a/b/c/" and you call pop that will return true which when inverted returns false, so you never get to the second pop and you end up in an infinite loop.

Comment (gregnazario) [reply crates/aptos/src/config/mod.rs:239]: Yeah this was my mistake good catch

**Review by gregnazario — APPROVED** [review]

**Pull request merged**